src/utils/TokenUtils.ts

```diff
@@ -115,15 +102,19 @@ export function isStablecoin(tokenSymbol: string): boolean {
   );
 }
 
-export function getTokenInfo(l2TokenAddress: string, chainId: number): L1Token {
+export function getTokenInfo(l2TokenAddress: string, chainId: number, tokenMapping = TOKEN_SYMBOLS_MAP): L1Token {
   // @dev This might give false positives if tokens on different networks have the same address. I'm not sure how
   // to get around this...
-  const tokenObject = Object.values(TOKEN_SYMBOLS_MAP).find(({ addresses }) => addresses[chainId] === l2TokenAddress);
+  let tokenObject = Object.values(tokenMapping).find(({ addresses }) => addresses[chainId] === l2TokenAddress);
   if (!tokenObject) {
     throw new Error(
       `TokenUtils#getTokenInfo: Unable to resolve token in TOKEN_SYMBOLS_MAP for ${l2TokenAddress} on chain ${chainId}`
     );
   }
+  if (chainIsL1(chainId)) {
```

Comment:
Why only do this for L1?

Reply:
We discussed offline, so for others, the idea is that when we resolve an L2 token address, we want to get the unique symbol (e.g. USDC.e or USDB), but for resolving L1 tokens, we want to get the "canonical" L1 token symbol.